experiments/download_ssl4eo.py

```diff
@@ -627,39 +497,42 @@ def fix_random_seeds(seed: int = 42) -> None:
     else:
         ext_path = os.path.join(args.save_path, "checked_locations.csv")
 
-    # build the grid from existing coordinates
-    grid_dict: Dict[Tuple[int, int], Any] = {}
-    if args.resume:
-        print("Load existing locations.")
-        for i, key in enumerate(tqdm(ext_coords.keys())):
-            c = ext_coords[key]
-            gridIndex = (math.floor(c[0] + 180), math.floor(c[1] + 90))
-            if gridIndex not in grid_dict.keys():
-                grid_dict[gridIndex] = {c}
-            else:
-                grid_dict[gridIndex].add(c)
+    # if match from pre-sampled coords (e.g. SSL4EO-S12)
+    if args.match_file:
+        match_coords = {}
+        with open(args.match_file) as csv_file:
+            reader = csv.reader(csv_file)
+            for row in reader:
+                key = row[0]
+                val1 = float(row[1])
+                val2 = float(row[2])
+                match_coords[key] = (val1, val2)  # lon, lat
```

Comment:
Where do these centroid coordinates get converted to the bbox to download? I don't see a size/radius parameter to the script.

Reply:
Here in the `get_patch_s2()` function for example:
```
    region = (
        ee.Geometry.Point(center_coord).buffer(radius).bounds()
    )  # sample region bound
    patch = image.select(*bands).sampleRectangle(region, defaultValue=0)
```
It samples a rectangle centered around the centroid of the bbox.